### PR TITLE
redisinsight: init at 2.30.0

### PR DESCRIPTION
**Pull request by gmemstr**
## Description of changes

Create the RedisInsight package (#242438).

While Redis provides AppImages and other packages, these are not well versioned, only containing the major version the binary was built for. Thus the only way to get proper versioning, from what I can tell, is to build from source.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -6318,6 +6318,12 @@
     githubId = 1447245;
     name = "Robin Gloster";
   };
+  gmemstr = {
+    email = "git@gmem.ca";
+    github = "gmemstr";
+    githubId = 1878840;
+    name = "Gabriel Simmer";
+  };
```
Comment (matthiasbeyer) [inline maintainers/maintainer-list.nix:6326]: This should be an individual commit.

**Review by matthiasbeyer — DISMISSED** [review]
Please change your commit message(s) to match the [commit message conventions of nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions)!

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (gmemstr) [reply maintainers/maintainer-list.nix:6326]: I was wondering about that. Will do.

**Comment by gmemstr** [comment]
PR for maintainership added in #252559, once merged I'll remove from this PR to prevent failures.

**Comment by gmemstr** [comment]
Realised it made more sense to place it under `pkgs/development/tools/redisinsight` instead of `pkgs/applications/misc`.

**Comment by gmemstr** [comment]
I think I've narrowed it down to the `__dirname` variable being set incorrectly, and I *think* a patch will solve the issue but not 100%.

**Comment by gmemstr** [comment]
Success! We have a building and running app.

**Comment by gmemstr** [comment]
As a sidenote, this could theoretically support all the platforms, but I don't have the ability to test at the moment, so sticking the with default x86 Linux.

**Comment by gmemstr** [comment]
@matthiasbeyer Don't mean to bug you but I'd love another review on this :)

**Comment by matthiasbeyer** [comment]
Well I don't think I am qualified to review yarn packaging, sorry.
But one thing I would say is that normally, new packages are not backported (to my knowledge)... besides that I cannot say much, sorry again!

File: pkgs/development/tools/redisinsight/default.nix [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    npm_config_node_gyp="${buildPackages.nodejs}/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" npm rebuild --verbose --sqlite=${sqlite.dev} sqlite3
+    popd
+
+    substituteInPlace redisinsight/api/config/default.ts \
+      --replace "process['resourcesPath']" "\"$out/share/redisinsight\"" \
+
+    runHook postConfigure
+  '';
+
+  dontPatchShebangs = true;
```
Comment (delroth) [inline pkgs/development/tools/redisinsight/default.nix:90]: This could use a comment explaining why that's necessary.

**Review by delroth — APPROVED** [review]
Disclaimer: I also don't know much about the yarn ecosystem in nixpkgs. But nothing in this PR seems particularly wrong to me, and in absence of a more qualified reviewer having a look I'd be happy to see this merged.

One tiny comment if you don't mind doing another change to this PR.

File: pkgs/development/tools/redisinsight/default.nix (hunk 2 above)
Comment (gmemstr) [reply pkgs/development/tools/redisinsight/default.nix:90]: I didn't actually test the build without this - I'll do so now and see if it breaks anything :)

Comment (gmemstr) [reply pkgs/development/tools/redisinsight/default.nix]: @delroth Built, and it works fine without this, so I've removed it.

**Comment by delroth** [comment]
(CI is now failing due to trailing whitespaces.)

**Comment by gmemstr** [comment]
Fixed trailing spaces.

**Comment by delroth** [comment]
Result of `nixpkgs-review pr 252557` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>redisinsight</li>
  </ul>
</details>

**Pull request merged**